crates/ruff_python_semantic/src/model.rs

```diff
@@ -766,13 +766,23 @@ impl<'a> SemanticModel<'a> {
             }
             BindingKind::ClassDefinition(_) | BindingKind::FunctionDefinition(_) => {
                 let value_name = UnqualifiedName::from_expr(value)?;
-                let resolved: QualifiedName = self
-                    .module_path?
-                    .iter()
-                    .map(String::as_str)
-                    .chain(value_name.segments().iter().copied())
-                    .collect();
-                Some(resolved)
+
+                // If we have a fully-qualified path for the module, use it.
+                if let Some(path) = self.module.path() {
+                    Some(
+                        path.iter()
+                            .map(String::as_str)
+                            .chain(value_name.segments().iter().copied())
+                            .collect(),
+                    )
+                } else {
+                    // Otherwise, if we're in (e.g.) a script, use the module name.
```

Comment:
I think the terminology is inconsistent which makes this change difficult to understand. For example, I don't understand when a file is a script? Sould we rename one of the variants in `ModuleSource` to script? or are the concepts independent from each other?

Reply:
Good call.

Reply:
Actually, they're somewhat independent (like `Script` and `Path` are odd as variants because they're not in the same category of concepts).